File: java_solutions/2600.k-items-with-the-maximum-sum.java
/*
 * @lc app=leetcode id=2600 lang=java
 *
 * [2600] K Items With the Maximum Sum
 */

class Solution {
    public int kItemsWithMaximumSum(int numOnes, int numZeros, int numNegOnes, int k) {
        return Math.min(k, numOnes) - Math.max(0, k - numOnes - numZeros);
    }
}

File: kotlin_solutions/2600.k-items-with-the-maximum-sum.kt
/*
 * @lc app=leetcode id=2600 lang=kotlin
 *
 * [2600] K Items With the Maximum Sum
 */

class Solution {fun kItemsWithMaximumSum(numOnes: Int, numZeros: Int, numNegOnes: Int, k: Int) = if (k <= numOnes) k else if (k <= numOnes + numZeros) numOnes else numOnes - minOf(k - numOnes - numZeros, numNegOnes)}
